src/darker/verification.py

```diff
@@ -67,3 +67,39 @@ def verify_ast_unchanged(
     except AssertionError as exc_info:
         debug_dump(black_chunks, edited_linenums)
         raise NotEquivalentError(str(exc_info))
+
+
+class ASTVerifier:
+    """
+    Verifies if reformatted TextDocument is AST-equivalent to baseline.
+
+    Keeps in-memory data about previous comparisons to improve performance.
+    """
+
+    def __init__(self, baseline: TextDocument) -> None:
+        self._baseline_ast_str = self._to_ast_str(baseline)
+        self._comparisons: Dict[str, bool] = {baseline.string: True}
+
+    @staticmethod
+    def _to_ast_str(document: TextDocument) -> str:
+        return "\n".join(stringify_ast(parse_ast(document.string)))
+
+    def is_equivalent_to_baseline(self, document: TextDocument) -> bool:
+        """Returns true if document is AST-equivalent to baseline"""
+        if document.string in self._comparisons:
+            return self._comparisons[document.string]
+
+        try:
+            document_ast_str = self._to_ast_str(document)
+        except SyntaxError:
+            comparison = False
+        else:
+            comparison = self._baseline_ast_str == document_ast_str
+
+        self._comparisons[document.string] = comparison
+        return self._comparisons[document.string]
+
+    def assert_equivalent_to_baseline(self, document: TextDocument) -> None:
```

Comment:
`ASTVerifier.assert_equivalent_to_baseline()` is unused (except in tests). Should probably be removed?

Reply:
I left it because I was unsure if we guarantee stability of importable API.
If not, of course this can be removed.

Reply:
I don't get it, there hasn't been a `darker.verification.ASTVerifier` class before, so there's no existing stable API to keep here.

But the question about guaranteeing Python API stability is a valid one of course. I opened #216 for that discussion, thanks for bringing it up @rogalski!